OSM data program V2.ipynb

```diff
@@ -0,0 +1,1086 @@
+    "    dropped_cols_df: copy of input df without dropped columns\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    dropped_cols_df = df.drop(col_names, inplace = False, axis = 1)\n",
+    "    return dropped_cols_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "3206d5bf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def rename_cols(df, col_dict):\n",
+    "    \"\"\"\n",
+    "    Function to rename column names of a dataframe\n",
+    "    \n",
+    "    Parameters\n",
+    "    __________\n",
+    "    df: dataframe to do operations on\n",
+    "    col_dict: dictionary of exisiting column names with their replacements\n",
+    "    \n",
+    "    Returns\n",
+    "    _______\n",
+    "    df: dataframe after renaming columns\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    df.rename(columns = col_dict, inplace = True)\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "0363eb90",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def convert_timestamp(df, col_name, time_unit):\n",
+    "    \"\"\"\n",
+    "    Function to convert timestamp from a long sequence of ints into datetime format\n",
+    "    \n",
+    "    Parameters\n",
+    "    __________\n",
+    "    df: dataframe to update\n",
+    "    col_name: name of the column to change\n",
+    "    time_unit: what type of time unit it should be, e.g. s for seconds\n",
+    "    \n",
+    "    Returns\n",
+    "    _______\n",
+    "    df: updated dataframe\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    df[col_name] = pd.to_datetime(df[col_name], unit = time_unit)\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "946d7170",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def df_to_file(df, file_name, encoding, folder):\n",
+    "    \"\"\"\n",
+    "    Function to create a file from the dataframe. The user specifies what file type it should be\n",
+    "    \n",
+    "    Parameters\n",
+    "    __________\n",
+    "    df: dataframe to save to file\n",
+    "    file_name: name of the new saved file\n",
+    "    encoding: what type of encoding should be used for the file if applicable\n",
+    "    folder: used to check whether the file already exists\n",
+    "    \n",
+    "    Returns\n",
+    "    _______\n",
+    "    exported_file: the newly created file\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    isExist = os.path.exists(folder + '/' + file_name)\n",
+    "    if isExist == True:\n",
+    "        print('This file already exists. Consider changing the name of the file')\n",
+    "    else:\n",
+    "        if '.csv' in file_name:\n",
+    "            exported_file = df.to_csv(file_name, encoding = encoding)\n",
+    "        elif '.parquet' in file_name:\n",
+    "            exported_file = df.to_parquet(file_name)\n",
+    "        return exported_file"
+   ]
+  },
```

Comment:
1) better to have override as an input argument
2) os.path.exists(folder + '/' + file_name) could cause issue when the os change, use pathlib 
3) the folder is not input/global argument
4) ".csv" in file_name could cause trouble, i.e. "test.csv.parquet". To check the postfix use the stdlib.